libfreerdp/core/nego.c

```diff
@@ -1303,31 +1321,49 @@ BOOL nego_send_negotiation_response(rdpNego* nego)
 				 * This also affects connections via port tunnels! (e.g. ssh -L)
 				 */
 				WLog_INFO(TAG, "Turning off encryption for local peer with standard rdp security");
-				settings->UseRdpSecurityLayer = FALSE;
-				settings->EncryptionLevel = ENCRYPTION_LEVEL_NONE;
+				if (!freerdp_settings_set_bool(settings, FreeRDP_UseRdpSecurityLayer, FALSE))
+					return FALSE;
+				if (!freerdp_settings_set_uint32(settings, FreeRDP_EncryptionLevel,
+				                                 ENCRYPTION_LEVEL_NONE))
+					return FALSE;
 			}
-			else if (!settings->RdpServerRsaKey && !settings->PrivateKeyFile &&
-			         !settings->PrivateKeyContent)
+			else if (!freerdp_settings_get_pointer(settings, FreeRDP_RdpServerRsaKey) &&
+			         !freerdp_settings_get_string(settings, FreeRDP_PrivateKeyFile) &&
+			         !freerdp_settings_get_string(settings, FreeRDP_PrivateKeyContent))
 			{
 				WLog_ERR(TAG, "Missing server certificate");
 				return FALSE;
 			}
 		}
-		else if (settings->SelectedProtocol == PROTOCOL_SSL)
+		else if (nego->SelectedProtocol == PROTOCOL_SSL)
 		{
-			settings->TlsSecurity = TRUE;
-			settings->NlaSecurity = FALSE;
-			settings->RdpSecurity = FALSE;
-			settings->UseRdpSecurityLayer = FALSE;
-			settings->EncryptionLevel = ENCRYPTION_LEVEL_NONE;
+			if (!freerdp_settings_set_bool(settings, FreeRDP_TlsSecurity, TRUE))
+				return FALSE;
+			if (!freerdp_settings_set_bool(settings, FreeRDP_NlaSecurity, FALSE))
+				return FALSE;
+			if (!freerdp_settings_set_bool(settings, FreeRDP_RdpSecurity, FALSE))
+				return FALSE;
+			if (!freerdp_settings_set_bool(settings, FreeRDP_UseRdpSecurityLayer, FALSE))
+				return FALSE;
+
+			if (!freerdp_settings_set_uint32(settings, FreeRDP_EncryptionLevel,
+			                                 ENCRYPTION_LEVEL_NONE))
+				return FALSE;
 		}
-		else if (settings->SelectedProtocol == PROTOCOL_HYBRID)
+		else if (nego->SelectedProtocol == PROTOCOL_HYBRID)
 		{
-			settings->TlsSecurity = TRUE;
-			settings->NlaSecurity = TRUE;
-			settings->RdpSecurity = FALSE;
-			settings->UseRdpSecurityLayer = FALSE;
-			settings->EncryptionLevel = ENCRYPTION_LEVEL_NONE;
+			if (!freerdp_settings_set_bool(settings, FreeRDP_TlsSecurity, TRUE))
```

Comment:
In that kind of case the usage of `freerdp_settings_set_XXX` makes it so heavy and useless (lots of unnecessary checks) compared to the original code...